### PR TITLE
[READY] Fixes for misc issues found by valgrind.

### PR DESCRIPTION
**Pull request by kevingranade**
Just some misc fixes found while running valgrind.

* morale_type:data::was_loaded was being used uninitialized.
* overmap::npcs was leaking.
* savegame_test was initializing overmap objects on the stack, which is bad because they're too large.
* Lots of unititialized members of various clsses.

Valgrind now reports no errors when running the default unit tests.

### CODE REVIEW

File: tests/savegame_test.cpp [hunk 1]
```diff
@@ -204,30 +204,34 @@ TEST_CASE("Reading a legacy overmap save.") {
     std::string legacy_save_name = "tests/data/legacy_0.C_overmap.sav";
     std::string new_save_name = "tests/data/jsionized_overmap.sav";
 
-    overmap test_map;
+    overmap *test_map = new overmap();
```
Comment (BevapDin) [inline tests/savegame_test.cpp]: Why a raw pointer and not a `std::unique_ptr`?

Comment (kevingranade) [inline tests/savegame_test.cpp]: No good reason, just not used to unipue ptrs.

File: src/game.cpp [hunk 2]
```diff
@@ -268,6 +269,9 @@ game::game() :
     tileset_zoom(16),
     weather_override( WEATHER_NULL )
 {
+    remoteveh_cache_turn = INT_MIN;
+    temperature = 0;
```
Comment (codemime) [inline src/game.cpp]: Why these two variables aren't in the initialization list like the others?

**Comment by codemime** [comment]
I guess you're planning to make valgrind checks a part of the build process. IMO that's a very good thing.

File: src/game.cpp (hunk 2 above)
Comment (kevingranade) [inline src/game.cpp]: Putting them in the initialization list requires you to put them in the correct order, and didn't feel like it.  If there's a reason to move it, I'd be happy to.

**Comment by kevingranade** [comment]
Yes, I'm planning on adding it to a travis build, have to see how long that makes it take though.

**Pull request merged**